<?xml version="1.0"?>
<launch>
    <!-- topic name -->
    <arg name="odom_topic_name" default="/roomba/odometry" />
    <arg name="map_topic_name" default="/map"/>
    <arg name="estimated_pose_topic_name" default="/estimated_pose"/>
    <arg name="estimated_poses_topic_name" default="/estimated_poses"/>
    <arg name="obj_topic_name" default="object_positions"/>
    <arg name="markers_topic_name" default="/markers"/>

    <!-- frame id -->
    <arg name="map_frame_id" default="map"/>
    <arg name="odom_frame_id" default="odom"/>
    <arg name="base_link_frame_id" default="base_link"/>
    <arg name="lidar_frame_id" default="scan"/>
    <arg name="thetas_frame_id" default="theta_s"/>
    <arg name="dynamixel_frame_id" default="dynamixel"/>
    <arg name="realsense_frame_id" default="realsense"/>

    <!-- map server -->
    <arg name="map_file" default="$(find mcl)/mapdata/map.yaml" />
    <node pkg="map_server" type="map_server" name="global_map" args="$(arg map_file)" />

    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find child_localization)/config/rviz/child_mcl.rviz" />

    <!-- child mcl -->
    <node pkg="child_localization" type="child_mcl_node" name="child_mcl_node">
        <param name="file_name" value="arrange_1.csv" />
        <param name="obj_poses_topic_name" value="$(arg obj_topic_name)" />
        <param name="estimated_pose_topic_name" value="$(arg estimated_pose_topic_name)" />
        <param name="estimated_poses_topic_name" value="$(arg estimated_poses_topic_name)" />
        <param name="map_topic_name" value="$(arg map_topic_name)" />
        <param name="map_frame_id" value="$(arg map_frame_id)" />
        <param name="odom_frame_id" value="$(arg odom_frame_id)" />
        <param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
        <rosparam command="load" file="$(find child_localization)/config/child_mcl/child_mcl.yaml" />
        <rosparam command="load" file="$(find child_localization)/config/object_list/object_list.yaml" />
    </node>

    <!-- tf -->
    <node pkg="child_localization" type="tf_broadcaster_node" name="tf_broadcaster_node" >
        <param name="is_odom_tf" value="true" />
        <param name="odom_topic_name" value="$(arg odom_topic_name)" />
        <param name="odom_frame_id" value="$(arg odom_frame_id)" />
        <param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
        <!--<param name="lidar_frame_id" value="$(arg lidar_frame_id)" />
        <param name="thetas_frame_id" value="$(arg thetas_frame_id)" />
        <param name="dynamixel_frame_id" value="$(arg dynamixel_frame_id)" />
        <param name="realsense_frame_id" value="$(arg realsense_frame_id)" />
        <rosparam command="load" file="$(find child_localization)/config/tf/tf.yaml" />-->
    </node>



</launch>